src/code.sh

```diff
@@ -16,33 +16,29 @@ chmod 600 ~/.ssh/id_rsa
 chmod 664 ~/.ssh/id_rsa.pub
 
 # Create variable containing options for connecting to genomics server via ssh.
-# -q suppresses warnings
-# StrictHostKeyChecking=no stops failed connections based on new unrecognised hosts
-# UserKnwonHostsFile=/dev/null uses a blank file for skip check of known host keys
-ssh_opts='-q -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i /home/dnanexus/.ssh/id_rsa'
+# -q suppresses warnings, -i indicates the SSH private key to use, -l presents username (required by rsync)
+# StrictHostKeyChecking=no; stops failed connections based on new unrecognised hosts
+# UserKnwonHostsFile=/dev/null; uses a blank file for skip check of known host keys
+ssh_opts='-q -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i /home/dnanexus/.ssh/id_rsa -l mokaguys'
 
 # Download old index.html for archiving
-# Note: Turning off StrictHostKeyChecking stops failure based on expected command line input
-scp $ssh_opts mokaguys@genomics.viapath.co.uk:/var/www/html/mokaguys/multiqc/index.html ${out_dir}/old_index.html
+# Rsync options: -a archive mode for recursive copy, -v verbose outputs, -h human-readable numbers,
+#   -z use compression in data transfer, -e SSH command with options
+rsync -avhz -e "ssh $ssh_opts" mokaguys@genomics.viapath.co.uk:/var/www/html/mokaguys/multiqc/index.html ${out_dir}/old_index.html
 
 # Copy HTML file to the server if it does not exist in the correct location
-file_exists_test="$ssh_opts mokaguys@genomics.viapath.co.uk test -e /var/www/html/mokaguys/multiqc/${multiqc_html_name}"
-if ! ssh ${file_exists_test}; then
+file_exists_test="$ssh_opts mokaguys@genomics.viapath.co.uk test -e /var/www/html/mokaguys/multiqc/reports/${multiqc_html_name}"
+if ssh ${file_exists_test}; then
    echo "File exists"
 else
-   scp $ssh_opts ${multiqc_html_path} mokaguys@genomics.viapath.co.uk:/var/www/html/mokaguys/multiqc/reports
+   rsync -avhz -e "ssh $ssh_opts" ${multiqc_html_path} mokaguys@genomics.viapath.co.uk:/var/www/html/mokaguys/multiqc/reports/${multiqc_html_name}
 fi
 
-# Get list of multiqc html files from server
-ssh $ssh_opts mokaguys@genomics.viapath.co.uk "ls -t /var/www/html/mokaguys/multiqc/reports/*.html" > multiqc_reports.txt
-
-# Install python HTML templating module
-sudo pip install Jinja2==2.10
 # Call python script to create a new index.html from html file list
-python update_index.py > ${out_dir}/new_index.html
+python update_index.py ${out_dir}/old_index.html ${multiqc_html_path} > ${out_dir}/new_index.html
```

Comment:
is this required if the file already exists? should this and below step be in the if loop?

Reply:
Thanks, added now